libr/include/r_heap_glibc.h

```diff
@@ -6,88 +6,183 @@
+	ut32	prev_size;	/* Size of previous chunk (if free).  */
+	ut32	size;       	/* Size in bytes, including overhead. */
+
+	ut32 fd;	        /* double links -- used only if free. */
+	ut32 bk;
+
+	/* Only used for large blocks: pointer to next larger size.  */
+	ut32 fd_nextsize; 	/* double links -- used only if free. */
+	ut32 bk_nextsize;
+} RHeapChunk32;
+
+/*
+typedef RHeapChunk64 *mfastbinptr64;
+typedef RHeapChunk64 *mchunkptr64;
+
+typedef RHeapChunk32 *mfastbinptr32;
+typedef RHeapChunk32 *mchunkptr32;
+*/
+
+typedef struct r_malloc_state_32 { 
+	int mutex; 				/* serialized access */ 
+	int flags; 				/* flags */
+	ut32 fastbinsY[NFASTBINS];		/* array of fastchunks */
+	ut32 top; 				/* top chunk's base addr */ 
+	ut32 last_remainder;			/* remainder top chunk's addr */
+	ut32 bins[NBINS * 2 - 2];   		/* array of remainder free chunks */
+	unsigned int binmap[BINMAPSIZE]; 	/* bitmap of bins */
+
+	ut32 next; 				/* double linked list of chunks */
+	ut32 next_free; 			/* double linked list of free chunks */
+
+	ut32 system_mem; 			/* current allocated memory of current arena */
+	ut32 max_system_mem;  			/* maximum system memory */
+} RHeap_MallocState32; 
+
+typedef struct r_malloc_state_64 { 
+	int mutex; 				/* serialized access */ 
+	int flags; 				/* flags */
+	ut64 fastbinsY[NFASTBINS];		/* array of fastchunks */
+	ut64 top; 				/* top chunk's base addr */ 
+	ut64 last_remainder;			/* remainder top chunk's addr */
+	ut64 bins[NBINS * 2 - 2];   		/* array of remainder free chunks */
+	unsigned int binmap[BINMAPSIZE]; 	/* bitmap of bins */
+
+	ut64 next; 				/* double linked list of chunks */
+	ut64 next_free; 			/* double linked list of free chunks */
+
+	ut64 system_mem;	 		/* current allocated memory of current arena */
+	ut64 max_system_mem;  			/* maximum system memory */
+} RHeap_MallocState64; 
+
+typedef struct r_heap_info_32 {
+	ut32 ar_ptr;			/* Arena for this heap. */
+	ut32 prev;		 	/* Previous heap. */
+	ut32 size;   			/* Current size in bytes. */
+	ut32 mprotect_size;		/* Size in bytes that has been mprotected PROT_READ|PROT_WRITE.  */
+
+	/* Make sure the following data is properly aligned, particularly
+	that sizeof (heap_info) + 2 * SZ is a multiple of
+	MALLOC_ALIGNMENT. */	
+	/* char pad[NPAD * SZ & MALLOC_ALIGN_MASK]; */
+} RHeapInfo32;
+
+typedef struct r_heap_info_64 {
+	ut64 ar_ptr;			/* Arena for this heap. */
+	ut64 prev;		 	/* Previous heap. */
+	ut64 size;   			/* Current size in bytes. */
+	ut64 mprotect_size;		/* Size in bytes that has been mprotected PROT_READ|PROT_WRITE.  */
 
 	/* Make sure the following data is properly aligned, particularly
-	that sizeof (heap_info) + 2 * SIZE_SZ is a multiple of
+	that sizeof (heap_info) + 2 * SZ is a multiple of
 	MALLOC_ALIGNMENT. */
-	char pad[NPAD * SIZE_SZ & MALLOC_ALIGN_MASK];
-} RHeapInfo;
+	/* char pad[NPAD * SZ & MALLOC_ALIGN_MASK]; */
+} RHeapInfo64;
+
+void update_main_arena_32(RCore *core, ut32 m_arena, RHeap_MallocState32 *main_arena);
```

Comment:
if it is public api it should be prefixed with R_API and should honor the naming conventions of the module where it belongs to.

imho all this stuff should be under r_debug_heap_*